### PR TITLE
clean up host operations

### PR DESCRIPTION
**Pull request by BerrevoetsRobbe**
**ID**
cc029ee

### CODE REVIEW

File: core/java/iesi-core/src/main/java/io/metadew/iesi/connection/HostConnection.java [hunk 1]
```diff
@@ -266,11 +266,11 @@ public ShellCommandResult executeRemoteCommandExec(String shellPath, String shel
                     HostConnection hostConnection = null;
                     if (i < jumphostConnections.length) {
                         jumphostConnection = jumphostConnections[i];
+                        String finalJumphostConnection = jumphostConnection;
```
Comment (hkhattabii) [inline core/java/iesi-core/src/main/java/io/metadew/iesi/connection/HostConnection.java:269]: Why assigning jumpHostConnection to finalJumpHostConnection ? I see in next lines that jumpHostConnection doesn't change, nothing change between jumpHostconnection and finalJumpHostConnectio. It's not redundant ?

File: core/java/iesi-core/src/main/java/io/metadew/iesi/connection/operation/ConnectionOperation.java [hunk 2]
```diff
@@ -70,15 +73,15 @@ public HostConnection getHostConnection(Connection connection) {
...
                     } else if (connectionTypeParameter.getKey().equalsIgnoreCase("temppath")) {
-                        tempPath = FrameworkCrypto.getInstance().decrypt(tempPath);
+                        tempPath = FrameworkCrypto.getInstance().decryptIfNeeded(tempPath);
                     }
                 }
             }
 
             hostConnection = new WindowsHostConnection(hostName, tempPath);
-        } else if (connection.getType().equalsIgnoreCase("host.linux")) {
+        } else if (connection.getType().equalsIgnoreCase("host.linux") || connection.getType().equalsIgnoreCase("host.unix")) {
```
Comment (hkhattabii) [inline core/java/iesi-core/src/main/java/io/metadew/iesi/connection/operation/ConnectionOperation.java:84]: Not better to do connection.getType().matches("host.linux|host.unix") for more readability ?

**Review by TawbaneAcc — APPROVED** [review]

File: core/java/iesi-core/src/main/java/io/metadew/iesi/connection/HostConnection.java (hunk 1 above)
Comment (BerrevoetsRobbe) [reply core/java/iesi-core/src/main/java/io/metadew/iesi/connection/HostConnection.java:269]: this is a requirement to use it in the stream on line 272

File: core/java/iesi-core/src/main/java/io/metadew/iesi/connection/operation/ConnectionOperation.java (hunk 2 above)
Comment (BerrevoetsRobbe) [reply core/java/iesi-core/src/main/java/io/metadew/iesi/connection/operation/ConnectionOperation.java:84]: not in my opinion, also we will have to include the case insensitivity in the regex pattern. A regex is more appropriate when the pattern is more complex or generic. Like yours for the 2XX check the OpenAPI PR

File: core/java/iesi-core/src/main/java/io/metadew/iesi/connection/HostConnection.java (hunk 1 above)
Comment (hkhattabii) [reply core/java/iesi-core/src/main/java/io/metadew/iesi/connection/HostConnection.java:269]: Oh okay I see

File: core/java/iesi-core/src/main/java/io/metadew/iesi/connection/operation/ConnectionOperation.java (hunk 2 above)
Comment (hkhattabii) [reply core/java/iesi-core/src/main/java/io/metadew/iesi/connection/operation/ConnectionOperation.java:84]: Indeed, my bad

**Review by hkhattabii — APPROVED** [review]

**Pull request merged**